### PR TITLE
Add to array (fix + unit tests)

### PR DESCRIPTION
**Pull request by KevinDockx**
Kirthi, this is the PR containing the changes for adding to an array (including unit tests)

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@KevinDockx__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>

TTYL, MSOTBOT;

**Comment by rynowak** [comment]
@GrabYourPitchforks

**Comment by kirthik** [comment]
@KevinDockx - Changes look fine to me but we need to fix spacing. I can fix that.

**Comment by KevinDockx** [comment]
@kirthik: tnx!  On the spacing issue: - ok, sounds good to me :)

**Comment by davidfowl** [comment]
Yah, we don't use tabs.

File: test/Microsoft.AspNet.JsonPatch.Test/SimpleObjectAdapterTests.cs [hunk 1]
```diff
@@ -130,6 +135,99 @@ public void AddToListInvalidPositionTooLargeWithSerialization()
...
+			};
+
+			// create patch
+			JsonPatchDocument<SimpleDTO> patchDoc = new JsonPatchDocument<SimpleDTO>();
+			patchDoc.Add<int>(o => o.IntegerList, 4, 3);
+
+			patchDoc.ApplyTo(doc);
+
+			Assert.Equal(new List<int>() { 1, 2, 3, 4 }, doc.IntegerList);
+
```
Comment (pranavkm) [inline test/Microsoft.AspNet.JsonPatch.Test/SimpleObjectAdapterTests.cs]: Remove empty lines.

File: test/Microsoft.AspNet.JsonPatch.Test/SimpleObjectAdapterTests.cs [hunk 2]
```diff
@@ -130,6 +135,99 @@ public void AddToListInvalidPositionTooLargeWithSerialization()
...
 		[Fact]
+		public void AddToListAtEnd()
+		{
+			var doc = new SimpleDTO()
+			{
+				IntegerList = new List<int>() { 1, 2, 3 }
+			};
+
+			// create patch
+			JsonPatchDocument<SimpleDTO> patchDoc = new JsonPatchDocument<SimpleDTO>();
```
Comment (pranavkm) [inline test/Microsoft.AspNet.JsonPatch.Test/SimpleObjectAdapterTests.cs]: Are we not using `var` convention in these files?

Comment (kirthik) [inline test/Microsoft.AspNet.JsonPatch.Test/SimpleObjectAdapterTests.cs]: We will. I will be cleaning up the code according to coding guidelines.

**Comment by kirthik** [comment]
Checked this into jsonpatch branch. This is related PR for jsonpatch branch - https://github.com/aspnet/Mvc/pull/2212

**Pull request closed without merging**